### PR TITLE
OSDOCS-3039: Improve kubelet CA update for Paused Machine Config Pools

### PR DESCRIPTION
**Pull request by mburke5678**
https://issues.redhat.com/browse/OSDOCS-3039

Previews
Added Unpaused note:
https://deploy-preview-45313--osdocs.netlify.app/openshift-enterprise/latest/security/compliance_operator/compliance-operator-remediation.html#compliance-applying_compliance-remediation

https://deploy-preview-45313--osdocs.netlify.app/openshift-enterprise/latest/support/troubleshooting/troubleshooting-operator-issues.html#troubleshooting-disabling-autoreboot-mco_troubleshooting-operator-issues

Added last sentence about alerts in the Unpaused note:
https://deploy-preview-45313--osdocs.netlify.app/openshift-enterprise/latest/updating/update-using-custom-machine-config-pools.html#update-using-custom-machine-config-pools-pause_update-using-custom-machine-config-pools

https://deploy-preview-45313--osdocs.netlify.app/openshift-enterprise/latest/updating/updating-cluster-within-minor.html#update-using-custom-machine-config-pools-canary_updating-cluster-within-minor

Added sentence about alerts in the Important note:
https://deploy-preview-45313--osdocs.netlify.app/openshift-enterprise/latest/architecture/control-plane.html#about-machine-config-operator_control-plane

https://deploy-preview-45313--osdocs.netlify.app/openshift-enterprise/latest/updating/update-using-custom-machine-config-pools.html

Added sentence about alerts in the note in Step 3:
https://deploy-preview-45313--osdocs.netlify.app/openshift-enterprise/latest/updating/update-using-custom-machine-config-pools.html#update-using-custom-machine-config-pools-about_update-using-custom-machine-config-pools

### CODE REVIEW

**Comment by mburke5678** [comment]
@jkyros Is this a PR that you can review? It is related to [MCO-77](https://issues.redhat.com/browse/MCO-77)

File: modules/update-using-custom-machine-config-pools-canary.adoc [hunk 1]
```diff
@@ -26,7 +26,9 @@ The rolling update process described in this topic involves:
...
+Pausing an MCP prevents the Machine Config Operator from applying any configuration changes on the associated nodes. Pausing an MCP also prevents any automatically-rotated certificates from being pushed to the associated nodes, including the automatic CA rotation of the `kube-apiserver-to-kubelet-signer` CA certificate. If the MCP is paused when the `kube-apiserver-to-kubelet-signer` CA certificate expires and the MCO attempts to automatically renew the certificate, the new certificate is created but not applied across the nodes in the respective machine config pool. This causes failure in multiple `oc` commands, including but not limited to `oc debug`, `oc logs`, `oc exec`, and `oc attach`. You receive alerts in the Alerting UI of the {product-title} web console if an MCP is paused when the certificates are rotated.
+
+ Pausing an MCP should be done with careful consideration about the `kube-apiserver-to-kubelet-signer` CA certificate expiration and for short periods of time only.
```
Comment (jkyros) [inline modules/update-using-custom-machine-config-pools-canary.adoc:31]: I think maybe the space at the beginning of this line is making it render that last sentence weirdly in its own box for this link:   [https://deploy-preview-45313--osdocs.netlify.app/openshift-enterprise/latest/updating/updating-cluster-within-minor.html#update-using-custom-machine-config-pools-canary_updating-cluster-within-minor]( https://deploy-preview-45313--osdocs.netlify.app/openshift-enterprise/latest/updating/updating-cluster-within-minor.html#update-using-custom-machine-config-pools-canary_updating-cluster-within-minor )

Everything else looks great!

Comment (jkyros) [inline modules/update-using-custom-machine-config-pools-canary.adoc:31]: I think maybe the space at the beginning of this line is making it render that last sentence weirdly in its own box for this link:   [https://deploy-preview-45313--osdocs.netlify.app/openshift-enterprise/latest/updating/updating-cluster-within-minor.html#update-using-custom-machine-config-pools-canary_updating-cluster-within-minor]( https://deploy-preview-45313--osdocs.netlify.app/openshift-enterprise/latest/updating/updating-cluster-within-minor.html#update-using-custom-machine-config-pools-canary_updating-cluster-within-minor )

Everything else looks great!

**Comment by mburke5678** [comment]
@rioliu-rh Please take a look. Essentially, the same sentence repeated in multiple locations.

File: modules/update-using-custom-machine-config-pools-about.adoc [hunk 2]
```diff
@@ -26,7 +26,7 @@ Do not remove the default worker label from the nodes. The nodes *must* have a r
...
+Pausing the MCP also pauses the kube-apiserver-to-kubelet-signer automatic CA certificates rotation. New CA certificates are generated at 292 days from the installation date and old certificates are removed 365 days from the installation date. See the link:https://access.redhat.com/articles/5651701[Understand CA cert auto renewal in Red Hat OpenShift 4] to find out how much time you have before the next automatic CA certificate rotation. Make sure the pools are unpaused when the CA cert rotation happens. If the MCPs are paused, the cert rotation does not happen, which causes the cluster to become degraded and causes failure in multiple `oc` commands, including but not limited to `oc debug`, `oc logs`, `oc exec`, and `oc attach`. You receive alerts in the Alerting UI of the {product-title} web console if an MCP is paused when the certificates are rotated.
```
Comment (rioliu-rh) [inline modules/update-using-custom-machine-config-pools-about.adoc:29]: > the cert rotation does not happen

same concern as above

File: modules/update-using-custom-machine-config-pools-pause.adoc [hunk 3]
```diff
@@ -9,7 +9,7 @@ In this canary rollout update process, after you label the nodes that you do not
...
+Pausing the MCP also pauses the kube-apiserver-to-kubelet-signer automatic CA certificates rotation. New CA certificates are generated at 292 days from the installation date and old certificates are removed 365 days from the installation date. See the link:https://access.redhat.com/articles/5651701[Understand CA cert auto renewal in Red Hat OpenShift 4] to find out how much time you have before the next automatic CA certificate rotation. Make sure the pools are unpaused when the CA cert rotation happens. If the MCPs are paused, the cert rotation does not happen, which causes the cluster to become degraded and causes failure in multiple `oc` commands, including but not limited to `oc debug`, `oc logs`, `oc exec`, and `oc attach`. You receive alerts in the Alerting UI of the {product-title} web console if an MCP is paused when the certificates are rotated.
```
Comment (rioliu-rh) [inline modules/update-using-custom-machine-config-pools-pause.adoc:12]: > the cert rotation does not happen

same concern as above

File: modules/compliance-applying.adoc [hunk 4]
```diff
@@ -16,6 +16,11 @@ After the Compliance Operator processes the applied remediation, the `status.App
 
 Note that when the Machine Config Operator applies a new `MachineConfig` object to nodes in a pool, all the nodes belonging to the pool are rebooted. This might be inconvenient when applying multiple remediations, each of which re-renders the composite `75-$scan-name-$suite-name` `MachineConfig` object. To prevent applying the remediation immediately, you can pause the machine config pool by setting the `.spec.paused` attribute of a `MachineConfigPool` object to `true`.
 
+[NOTE]
+====
+Make sure the pools are unpaused when the CA cert rotation happens. If the MCPs are paused, the cert rotation does not happen, which causes the cluster to become degraded and causes failure in multiple `oc` commands, including but not limited to `oc debug`, `oc logs`, `oc exec`, and `oc attach`. You receive alerts in the Alerting UI of the {product-title} web console if an MCP is paused when the certificates are rotated.
```
[the remark this answers is not in the document]
Comment (jkyros) [reply modules/compliance-applying.adoc:21]: What @rioliu-rh  is saying is technically correct. As far as the nodes are concerned, the "upstream" certificate rotation does happen, and the certificates but it doesn't "happen" for the nodes. Whether we want to be that specific/granular in terms of what "happens" where,I don't know?

[the remark this answers is not in the document]
Comment (mburke5678) [reply modules/compliance-applying.adoc:21]: @rioliu-rh @jkyros Is this more accurate?
_Make sure the pools are unpaused when the CA cert rotation happens. If the MCPs are paused, **the MCO cannot push the newly-rotated certs to those nodes**,_

[the remark this answers is not in the document]
Comment (rioliu-rh) [reply modules/compliance-applying.adoc]: yes, looks good to me

**Review by sergiordlr — COMMENTED** [review]
Looks good to me.

File: modules/compliance-applying.adoc [hunk 5]
```diff
@@ -16,6 +16,11 @@ After the Compliance Operator processes the applied remediation, the `status.App
 
 Note that when the Machine Config Operator applies a new `MachineConfig` object to nodes in a pool, all the nodes belonging to the pool are rebooted. This might be inconvenient when applying multiple remediations, each of which re-renders the composite `75-$scan-name-$suite-name` `MachineConfig` object. To prevent applying the remediation immediately, you can pause the machine config pool by setting the `.spec.paused` attribute of a `MachineConfigPool` object to `true`.
 
+[NOTE]
+====
+Make sure the pools are unpaused when the CA cert rotation happens. If the MCPs are paused, the MCO cannot push the newly-rotated certs to those nodes. This causes the cluster to become degraded and causes failure in multiple `oc` commands, including but not limited to `oc debug`, `oc logs`, `oc exec`, and `oc attach`. You receive alerts in the Alerting UI of the {product-title} web console if an MCP is paused when the certificates are rotated.
```
Comment (jeana-redhat) [inline modules/compliance-applying.adoc:21]: Technically, "including but not limited to" should have commas to set it off the parenthetical part ("including, but not limited to,"), but that would yield: "multiple `oc` commands, including, but not limited to, `oc debug`, `oc logs`, `oc exec`, and `oc attach`" which imo is pretty clunky. 

Might be simplest/most minimalist to just let "including: do the work as below:

```suggestion
Make sure the pools are unpaused when the CA certificate rotation happens. If the MCPs are paused, the MCO cannot push the newly rotated certificates to those nodes. This causes the cluster to become degraded and causes failure in multiple `oc` commands, including `oc debug`, `oc logs`, `oc exec`, and `oc attach`. You receive alerts in the Alerting UI of the {product-title} web console if an MCP is paused when the certificates are rotated.
```

Either way NB s/`newly-rotated`/`newly rotated` which is easy to miss in the little diff preview :slightly_smiling_face:

File: modules/troubleshooting-disabling-autoreboot-mco.adoc [hunk 6]
```diff
@@ -24,7 +24,9 @@ To avoid unwanted disruptions, you can modify the machine config pool (MCP) to p
...
+Pausing an MCP prevents the MCO from applying any configuration changes on the associated nodes. Pausing an MCP also prevents any automatically-rotated certificates from being pushed to the associated nodes, including the automatic rotation of the `kube-apiserver-to-kubelet-signer` CA certificate. If the MCP is paused when the `kube-apiserver-to-kubelet-signer` CA certificate expires, and the MCO attempts to renew the certificate automatically, the MCO cannot push the newly-rotated certs to those nodes. This causes the cluster to become degraded and causes failure in multiple `oc` commands, including but not limited to `oc debug`, `oc logs`, `oc exec`, and `oc attach`. You receive alerts in the Alerting UI of the {product-title} web console if an MCP is paused when the certificates are rotated.
```
Comment (jeana-redhat) [inline modules/troubleshooting-disabling-autoreboot-mco.adoc:27]: Might also consider breaking "If the MCP is paused when..." onto a new line for a bit easier readability.

```suggestion
Pausing an MCP prevents the MCO from applying any configuration changes on the associated nodes. Pausing an MCP also prevents any automatically rotated certificates from being pushed to the associated nodes, including the automatic rotation of the `kube-apiserver-to-kubelet-signer` CA certificate. If the MCP is paused when the `kube-apiserver-to-kubelet-signer` CA certificate expires, and the MCO attempts to renew the certificate automatically, the MCO cannot push the newly rotated certificates to those nodes. This causes the cluster to become degraded and causes failure in multiple `oc` commands, including `oc debug`, `oc logs`, `oc exec`, and `oc attach`. You receive alerts in the Alerting UI of the {product-title} web console if an MCP is paused when the certificates are rotated.
```

File: modules/understanding-machine-config-operator.adoc [hunk 7]
```diff
@@ -46,6 +46,8 @@ When changes are made to a machine configuration, the Machine Config Operator (M
 
 To prevent the nodes from automatically rebooting after machine configuration changes, before making the changes, you must pause the autoreboot process by setting the `spec.paused` field to `true` in the corresponding machine config pool. When paused, machine configuration changes are not applied until you set the `spec.paused` field to `false` and the nodes have rebooted into the new configuration.
 
+Make sure the pools are unpaused when the CA cert rotation happens. If the MCPs are paused, the MCO cannot push the newly-rotated certs to those nodes. This causes the cluster to become degraded and causes failure in multiple `oc` commands, including but not limited to `oc debug`, `oc logs`, `oc exec`, and `oc attach`. You receive alerts in the Alerting UI of the {product-title} web console if an MCP is paused when the certificates are rotated.
```
Comment (jeana-redhat) [inline modules/understanding-machine-config-operator.adoc:49]:
```suggestion
Make sure the pools are unpaused when the CA certificate rotation happens. If the MCPs are paused, the MCO cannot push the newly rotated certificates to those nodes. This causes the cluster to become degraded and causes failure in multiple `oc` commands, including `oc debug`, `oc logs`, `oc exec`, and `oc attach`. You receive alerts in the Alerting UI of the {product-title} web console if an MCP is paused when the certificates are rotated.
```

File: modules/update-using-custom-machine-config-pools-about.adoc [hunk 8]
```diff
@@ -26,7 +26,7 @@ Do not remove the default worker label from the nodes. The nodes *must* have a r
...
+Pausing the MCP also pauses the kube-apiserver-to-kubelet-signer automatic CA certificates rotation. New CA certificates are generated at 292 days from the installation date and old certificates are removed 365 days from the installation date. See the link:https://access.redhat.com/articles/5651701[Understand CA cert auto renewal in Red Hat OpenShift 4] to find out how much time you have before the next automatic CA certificate rotation. Make sure the pools are unpaused when the CA cert rotation happens. If the MCPs are paused, the MCO cannot push the newly-rotated certs to those nodes. This causes the cluster to become degraded and causes failure in multiple `oc` commands, including but not limited to `oc debug`, `oc logs`, `oc exec`, and `oc attach`. You receive alerts in the Alerting UI of the {product-title} web console if an MCP is paused when the certificates are rotated.
```
Comment (jeana-redhat) [inline modules/update-using-custom-machine-config-pools-about.adoc:29]: Also suggest breaking "Make sure the pools..." into a new paragraph here.

```suggestion
Pausing the MCP also pauses the `kube-apiserver-to-kubelet-signer` automatic CA certificates rotation. New CA certificates are generated 292 days from the installation date and old certificates are removed 365 days from the installation date. See link:https://access.redhat.com/articles/5651701[Understand CA cert auto renewal in Red Hat OpenShift 4] to find out how much time you have before the next automatic CA certificate rotation. Make sure the pools are unpaused when the CA certificate rotation happens. If the MCPs are paused, the MCO cannot push the newly rotated certificates to those nodes. This causes the cluster to become degraded and causes failure in multiple `oc` commands, including `oc debug`, `oc logs`, `oc exec`, and `oc attach`. You receive alerts in the Alerting UI of the {product-title} web console if an MCP is paused when the certificates are rotated.
```

File: modules/update-using-custom-machine-config-pools-canary.adoc [hunk 9]
```diff
@@ -26,7 +26,9 @@ The rolling update process described in this topic involves:
...
+Pausing an MCP prevents the Machine Config Operator from applying any configuration changes on the associated nodes. Pausing an MCP also prevents any automatically-rotated certificates from being pushed to the associated nodes, including the automatic CA rotation of the `kube-apiserver-to-kubelet-signer` CA certificate. If the MCP is paused when the `kube-apiserver-to-kubelet-signer` CA certificate expires and the MCO attempts to automatically renew the certificate, the new certificate is created but not applied across the nodes in the respective machine config pool. This causes failure in multiple `oc` commands, including but not limited to `oc debug`, `oc logs`, `oc exec`, and `oc attach`. You receive alerts in the Alerting UI of the {product-title} web console if an MCP is paused when the certificates are rotated.
```
Comment (jeana-redhat) [inline modules/update-using-custom-machine-config-pools-canary.adoc:29]: Same suggestions as in https://github.com/openshift/openshift-docs/pull/45313/files#diff-7c4392ac3894b8af0cb70e0770dfec2ace9c0e650d162d278172c7325e2d62c9

File: modules/update-using-custom-machine-config-pools-pause.adoc [hunk 10]
```diff
@@ -9,7 +9,7 @@ In this canary rollout update process, after you label the nodes that you do not
...
+Pausing the MCP also pauses the kube-apiserver-to-kubelet-signer automatic CA certificates rotation. New CA certificates are generated at 292 days from the installation date and old certificates are removed 365 days from the installation date. See the link:https://access.redhat.com/articles/5651701[Understand CA cert auto renewal in Red Hat OpenShift 4] to find out how much time you have before the next automatic CA certificate rotation. Make sure the pools are unpaused when the CA cert rotation happens. If the MCPs are paused, the MCO cannot push the newly-rotated certs to those nodes. This causes the cluster to become degraded and causes failure in multiple `oc` commands, including but not limited to `oc debug`, `oc logs`, `oc exec`, and `oc attach`. You receive alerts in the Alerting UI of the {product-title} web console if an MCP is paused when the certificates are rotated.
```
Comment (jeana-redhat) [inline modules/update-using-custom-machine-config-pools-pause.adoc:12]: Same suggestions as in https://github.com/openshift/openshift-docs/pull/45313/files#diff-e1aa84e22ab6b1a3a39c12c0b090062bb68d38304f6b250aeccfc9d4864c97bd

File: updating/update-using-custom-machine-config-pools.adoc [hunk 11]
```diff
@@ -28,7 +28,9 @@ This scenario has not been tested and might result in an undefined cluster state
...
+Pausing a machine config pool prevents the Machine Config Operator from applying any configuration changes on the associated nodes. Pausing an MCP also prevents any automatically-rotated certificates from being pushed to the associated nodes, including the automatic CA rotation of the `kube-apiserver-to-kubelet-signer` CA certificate. If the MCP is paused when the `kube-apiserver-to-kubelet-signer` CA certificate expires and the MCO attempts to automatically renew the certificate, the new certificate is created but not applied across the nodes in the respective machine config pool. This causes failure in multiple `oc` commands, including but not limited to `oc debug`, `oc logs`, `oc exec`, and `oc attach`. You receive alerts in the Alerting UI of the {product-title} web console if an MCP is paused when the certificates are rotated.
```
Comment (jeana-redhat) [inline updating/update-using-custom-machine-config-pools.adoc:31]: Same suggestions as in https://github.com/openshift/openshift-docs/pull/45313/files#diff-7c4392ac3894b8af0cb70e0770dfec2ace9c0e650d162d278172c7325e2d62c9

**Review by jeana-redhat — COMMENTED** [review]
Nice! This is pretty complex but very well-explained. Left a few comments that I think might make it easier to read.

**Pull request merged**